### ISSUE TITLE
Support webhook default configuration from Slack UI

### ISSUE THREAD

**Issue by cknowles** [rackspace-cookbooks/chef-slack_handler#16]
Slack supports default configuration for icon and name of a webhook.

Inside (slack_handler_webhook.rb)[https://github.com/rackspace-cookbooks/chef-slack_handler/blob/master/files/default/slack_handler_webhook.rb] could the defaults be removed so the defaults built into Slack take effect instead? So remove the `|| ':fork_and_knife:'` and `|| 'chef_handler'` parts from this block:

```
  def initialize(config = {})
    # set defaults for any missing attributes
    @config = config
    @timeout = @config[:timeout] || 15
    @icon_emoji = @config[:icon_emoji]
    @username = @config[:username]
    @webhooks = @config[:webhooks]
  end
```

**Comment by nicwaller** [comment]
I agree that it would be preferable to be able to use the icon and username defined in the Slack integration, rather than always overriding it.

But it's not enough to fix this in `initialize`. If the username isn't set to a sane default then it will be nil, and the resultant JSON sent to Slack will look like this.

``` json
{
  "username": null,
  "etc": "etc"
}
```

So the `request_body` def needs to be patched too, because that's where the JSON request gets built. If config[:username] is nil, it shouldn't be included in the request at all.

**Comment by cknowles** [comment]
Just working on a pull request for this.

**Comment by cknowles** [comment]
This actually works with null username by the way, Slack still uses the default set on its side if null is submitted. Will patch it anyway.

**Comment by martinb3** [comment]
Fixed in #19.
